### PR TITLE
Reindex: Use request flavored methods

### PR DESCRIPTION
**Pull request by nik9000**
Use the new request flavored methods for the low level rest client
introduced in #29623 in reindex.

### CODE REVIEW

**Review by javanna — APPROVED** [review]

File: modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java [hunk 1]
```diff
@@ -160,10 +152,11 @@ static HttpEntity initialSearchEntity(SearchRequest searchRequest, BytesReferenc
 
             entity.endObject();
             BytesRef bytes = BytesReference.bytes(entity).toBytesRef();
-            return new ByteArrayEntity(bytes.bytes, bytes.offset, bytes.length, ContentType.APPLICATION_JSON);
+            request.setEntity(new ByteArrayEntity(bytes.bytes, bytes.offset, bytes.length, ContentType.APPLICATION_JSON));
```
Comment (javanna) [inline modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java]: question: should we use `NByteArrayEntity` instead?

File: modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java [hunk 2]
```diff
@@ -193,45 +186,50 @@ private static String sortToUri(SortBuilder<?> sort) {
...
         }
+
         try (XContentBuilder entity = JsonXContent.contentBuilder()) {
-            return new StringEntity(Strings.toString(entity.startObject()
-                .field("scroll_id", scroll)
-                .endObject()), ContentType.APPLICATION_JSON);
+            entity.startObject()
+                    .field("scroll_id", scroll)
+                .endObject();
+            request.setEntity(new StringEntity(Strings.toString(entity), ContentType.APPLICATION_JSON));
```
Comment (javanna) [inline modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java]: should we use `NStringEntity` instead? Which makes me wonder, should we make this easier for users, and expose   `setEntityAsString` and `setEntityAsByteArray` methods additionally to what we expose now??

File: modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java (hunk 1 above)
Comment (nik9000) [reply modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java]: I dunno! I've never really into the difference.

File: modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java (hunk 2 above)
Comment (nik9000) [reply modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java]: Hmmm. I think I'll merge as is and make a followup with those so we can talk about them!

**Comment by nik9000** [comment]
Thanks for reviewing @javanna! I've merged to master and backported to 6.x.

**Pull request merged**